mlir/test/mlir-miopen-driver/e2e/conv2d_harness_cpu_verifier_kyxc_nhwc_nhwk.mlir

```diff
@@ -0,0 +1,193 @@
+// RUN: mlir-miopen-driver -p -fil_layout=kyxc -in_layout=nhwc -out_layout=nhwk --host %s -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=E2E
+
+// filter: KYXC
+// input : NHWC
+// output: NHWK
+module {
+  func @main() {
+    // allocate CPU memory for gpu_conv.
+    %0 = alloc() : memref<128x3x3x8xf32>
+    %1 = alloc() : memref<128x32x32x8xf32>
+    %2 = alloc() : memref<128x30x30x128xf32>
+
+    %3 = memref_cast %0 : memref<128x3x3x8xf32> to memref<?x?x?x?xf32>
+    %4 = memref_cast %1 : memref<128x32x32x8xf32> to memref<?x?x?x?xf32>
+    %5 = memref_cast %2 : memref<128x30x30x128xf32> to memref<?x?x?x?xf32>
+
+    // populate initial values.
+    %cst = constant 1.000000e+00 : f32
+    %cst_0 = constant 0.000000e+00 : f32
+    call @mcpuMemset4DFloat(%3, %cst) : (memref<?x?x?x?xf32>, f32) -> ()
+    call @mcpuMemset4DFloat(%4, %cst) : (memref<?x?x?x?xf32>, f32) -> ()
+    call @mcpuMemset4DFloat(%5, %cst_0) : (memref<?x?x?x?xf32>, f32) -> ()
+
+    // launch gpu convolution
+    call @gpu_conv(%0, %1, %2) : (memref<128x3x3x8xf32>, memref<128x32x32x8xf32>, memref<128x30x30x128xf32>) -> ()
+
+    // allocate CPU memory for cpu_conv and populate initial values.
+    %6 = alloc() : memref<128x3x3x8xf32>
+    %7 = memref_cast %6 : memref<128x3x3x8xf32> to memref<?x?x?x?xf32>
+    call @mcpuMemset4DFloat(%7, %cst) : (memref<?x?x?x?xf32>, f32) -> ()
+
+    %8 = alloc() : memref<128x32x32x8xf32>
+    %9 = memref_cast %8 : memref<128x32x32x8xf32> to memref<?x?x?x?xf32>
+    call @mcpuMemset4DFloat(%9, %cst) : (memref<?x?x?x?xf32>, f32) -> ()
+
+    %10 = alloc() : memref<128x30x30x128xf32>
+    %11 = memref_cast %10 : memref<128x30x30x128xf32> to memref<?x?x?x?xf32>
+    call @mcpuMemset4DFloat(%11, %cst_0) : (memref<?x?x?x?xf32>, f32) -> ()
+
+    // launch cpu convolution
+    call @conv2d_host(%6, %8, %10) : (memref<128x3x3x8xf32>, memref<128x32x32x8xf32>, memref<128x30x30x128xf32>) -> ()
+
+    // verity results
+    call @verify_results(%10, %2) : (memref<128x30x30x128xf32>, memref<128x30x30x128xf32>) -> ()
```

Comment:
Shall we use `-ph` instead of manually populating those?

Reply:
This test was created deliberately to skip `-ph`. In fact it was the source how `ph` was created.